update_branch_protections.py

```diff
@@ -0,0 +1,234 @@
+    prot_master: bool = False,
+    prot_pages: bool = False,
+    prot_default: bool = False,
+    repo_data_path: str = "",
+    write: bool = False
+):
+    if repo_data_path:
+        # run for each repo in list
+        repo_data = parse_repo_list(repo_data_path)
+        for protection_group in repo_data:
+            protection_group.apply_protections(write=write)
+        return
+
+    if not repo_name:
+        return
+
+    # apply settings for a single repo
+    protection_group = ProtectionGroup(
+        owner=owner, repo_name=repo_name, repo_type=repo_type,
+        master=prot_master, gh_pages=prot_pages, default=prot_default
+    )
+    protection_group.apply_protections(write=write)
+
+    return
+
+
+def _create_argparser() -> argparse.ArgumentParser:
+    """
+    Create an ArgumentParser for update_branch_protections
+
+    Returns
+    -------
+    argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser()
+    # specify an owner, repo, and settings
+    parser.add_argument("owner", type=str, default='pcdshub', nargs='?',
+                        help='Organization or owner of the repository, "pcdshub"'
+                             'by default')
+    parser.add_argument("repo_name", type=str, default='', nargs='?',
+                        help='Name of the repository')
+    parser.add_argument("repo_type", type=str, default='Other', nargs='?',
+                        help='Type of the repository.  Should be one of '
+                             f'{list(ProtectionGroup.CHECK_NAME_MAP.keys())}')
+    parser.add_argument("--protect-master", action="store_true", dest='prot_master',
+                        help="Add master protection.  This should be applied as "
+                             "much as possible. This rule requires pull requests and "
+                             "disallows force pushes. This protection also adds required "
+                             "status checks depending on the repo_type")
+    parser.add_argument("--protect-pages", action="store_true", dest='prot_pages',
+                        help="Add gh-pages branch protection, allowing force pushes "
+                             "to the gh-pages branch specifically")
+    parser.add_argument("--protect-default", action="store_true", dest='prot_default',
+                        help="Add default (*) branch protection, disallows "
+                             "the creation of new branches")
+
+    # Optionally specify everything at once
+    parser.add_argument("--repo-data-path", type=str, dest='repo_data_path',
+                        help='Path to repo data csv.  Expects columns for: '
+                             f'{[f.name for f in fields(ProtectionGroup)]}')
+
+    parser.add_argument("--write", action="store_true", dest="write")
+
+    return parser
+
+
+def _main(args=None):
+    """CLI entrypoint."""
+    parser = _create_argparser()
+    return main(**vars(parser.parse_args(args=args)))
```

Comment:
This is a pretty concise way of consuming the args, I've never seen it before. I like it.

Reply:
You can thank Ken for that I shamelessly lifted this from him

Reply:
I guess I probably have seen it before then, and the information leaked out of my skull!